### ISSUE TITLE
[Bibdata/production] Redis::CommandError: MISCONF Redis is configured to save RDB snapshots, but is currently not able to persist on disk. Commands that may modify the data set are disabled. Please check Redis logs for details about the error.

### ISSUE THREAD

**Issue by christinach** [pulibrary/marc_liberation#500]
## Backtrace

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/54497/faults/37474446)

**Comment by christinach** [comment]
It seems an issue with redis permissions on the disk in bibdata-worker2.

**Comment by christinach** [comment]
```
1255:M 14 Oct 02:55:45.157 * Background saving terminated with success
1255:M 14 Oct 03:00:46.044 * 10 changes in 300 seconds. Saving...
1255:M 14 Oct 03:00:46.069 # Can't save in background: fork: Cannot allocate memory
1255:M 14 Oct 03:00:52.039 * 10 changes in 300 seconds. Saving...
1255:M 14 Oct 03:00:52.043 # Can't save in background: fork: Cannot allocate memory
1255:M 14 Oct 03:00:59.658 * 10 changes in 300 seconds. Saving...
1255:M 14 Oct 03:01:02.670 # Can't save in background: fork: Cannot allocate memory
1255:M 14 Oct 03:01:06.427 * 10 changes in 300 seconds. Saving...
1255:M 14 Oct 03:01:06.427 # Can't save in background: fork: Cannot allocate memory
1255:M 14 Oct 03:01:12.053 * 10 changes in 300 seconds. Saving...
1255:M 14 Oct 03:01:12.053 # Can't save in background: fork: Cannot allocate memory
1255:M 14 Oct 03:01:18.036 * 10 changes in 300 seconds. Saving...
```

**Comment by christinach** [comment]
or maybe a memory issue.

**Comment by christinach** [comment]
https://redis.io/topics/faq#background-saving-fails-with-a-fork-error-under-linux-even-if-i-have-a-lot-of-free-ram

**Comment by christinach** [comment]
Will update overcommit_memory to 1 based on the Redis documentation. @kayiwa is this ok ?

**Comment by christinach** [comment]
```
/proc/sys/vm/overcommit_memory
              This file contains the kernel virtual memory accounting mode.
              Values are:

                     0: heuristic overcommit (this is the default)
                     1: always overcommit, never check
                     2: always check, never overcommit

              In mode 0, calls of mmap(2) with MAP_NORESERVE are not
              checked, and the default check is very weak, leading to the
              risk of getting a process "OOM-killed".

              In mode 1, the kernel pretends there is always enough memory,
              until memory actually runs out.  One use case for this mode is
              scientific computing applications that employ large sparse
              arrays.  In Linux kernel versions before 2.6.0, any nonzero
              value implies mode 1.

              In mode 2 (available since Linux 2.6), the total virtual
              address space that can be allocated (CommitLimit in /proc/mem‐
              info) is calculated as

                  CommitLimit = (total_RAM - total_huge_TLB) *
                                overcommit_ratio / 100 + total_swap

              where:

                   *  total_RAM is the total amount of RAM on the system;

                   *  total_huge_TLB is the amount of memory set aside for
                      huge pages;

                   *  overcommit_ratio is the value in /proc/sys/vm/overcom‐
                      mit_ratio; and

                   *  total_swap is the amount of swap space.

              For example, on a system with 16GB of physical RAM, 16GB of
              swap, no space dedicated to huge pages, and an overcom‐
              mit_ratio of 50, this formula yields a CommitLimit of 24GB.

              Since Linux 3.14, if the value in /proc/sys/vm/overcom‐
              mit_kbytes is nonzero, then CommitLimit is instead calculated
              as:

                  CommitLimit = overcommit_kbytes + total_swap

              See also the description of /proc/sys/vm/admiin_reserve_kbytes
              and /proc/sys/vm/user_reserve_kbytes.
```
